Framework/PythonInterface/plugins/algorithms/SetSampleFromLogs.py

```diff
@@ -164,12 +166,17 @@ def PyExec(self):
         environment = self._createEnvironment(runObject, instrEnum)
 
         # let SetSample generate errors if anything is wrong
-        SetSample(InputWorkspace=wksp,
-                  Material=material,
-                  Geometry=geometry,
-                  Environment=environment,
-                  ContainerGeometry=geometryContainer,
-                  ContainerMaterial=materialContainer)
+        if self.getProperty("Environment").value.values():
```

Comment:
```suggestion
        if environment.values():
```
but I think this check should be dropped as `_createEnvironment()` should correctly return an empty value when appropriate.